sdk/avalon_sdk/connector/blockchains/fabric/fabric_work_order.py

```diff
@@ -119,33 +119,35 @@ def handle_fabric_event(event, block_num, txn_id, status):
             payload = event['payload'].decode("utf-8")
             resp = json.loads(payload)
             response = json.loads(resp["workOrderResponse"])
+            # to use global scope of wo_resp
+            nonlocal wo_resp
```

Comment:
Can't we directly use the response in events payload ?. Any specific reason why we need to use another variable with global scope ?  How are the changes in PR aligned with overall design of making event handling common across Fabric and Ethereum ?

Reply:
Function handle_fabric_event() nested function inside the work_order_get_result(), we can't return result from there because this handler function called from other places. we have to use variable in the outer function, and update the variable in the nested function. These changes are specific to work_order_get result().

To make event handling common across Fabric and Ethereum, we have to come up with apis in the interface to create event handler.

Reply:
if `handle_fabric_event()` is called from other places then why do we need to make this function nested since the scope of the function is not limited to `work_order_get_result()`. Also `wo_resp` variable in `handle_fabric_event()` makes sense only if it's used within `work_order_get_result()`.  

If you plan to modify it as part of unifying event handling for Fabric and Ethereum then  separate PR can be raised to handle this as well

Reply:
Scope of the function handle_fabric_event() is nested within the work_order_get_result(). Instead of defining the function out side, within the class it is good to have nested function. Do you see any difficulties in that? To know more about nested function go through this - https://stackabuse.com/python-nested-functions
Variable wo_resp variable is used in the work_order_get_result() it local to both the functions.  

This how fabric events handled with callback handler function. This change is not unify event handling approach.

Reply:
This approach we can't unify it because we created separate implementation class to handle it based the sdk what is provided the blockchain.

Reply:
I'm fine with nested function here as long as scope of nested function is within outer function `work_order_get_result()`. Is that the case now ? I'm confused with your above comment ---.> "because this handler function called from other places"

Reply:
Are we planning to raise a separate PR define common interface for event handling.  Specific blockchain SDK API calls will be implementation specific. But can't we make asyncio constructs for handling events common?.

Reply:
> I'm fine with nested function here as long as scope of nested function is within outer function `work_order_get_result()`. Is that the case now ? I'm confused with your above comment ---.> "because this handler function called from other places"

variable scope will be within the function work_order_get_result(). I meant Handler function will be called once we get the event, we are not calling it explicitly.